### ISSUE TITLE
Panic at break in an infinte loop, in finally block

### ISSUE THREAD

**Issue by MahnurA** [boa-dev/boa#3054]
**Describe the bug**
In a try-catch-finally -- in the finally block, panic in a one line break statement after an infinite loop, _if there are no braces around the break statement._ 

**To Reproduce**
These JavaScripts crash with `thread 'main' panicked at 'stack was empty', boa_engine/src/vm/opcode/control_flow/finally.rs:137:59`
```js
try {
}
catch {
} finally {
 while(true) 
  break;
}
```
or 

```js
try {
}
catch {
} finally {
 for(;;)
  break;
}
```

**Expected behavior**
Should not panic. Javascript allows one-line statements after a condition to not be encompassed by braces.

**Build environment:**
• OS: Ubuntu
• Version: 20.04
• Target triple: x86_64-unknown-linux-gnu
• Rustc version: rustc 1.70.0

**Additional context**
Note that this Javascript does _not_ panic, i.e. if braces are present. It is also clearly breaking out of the infinite loop as "in finally" does get printed. 
```js
try {
}
catch {
} finally {
 while(true) {
  break;
 }
 console.log("in finally")
}
```

**Comment by dirkdev98** [comment]
I'm sorry for not commenting on this before working on this, will make sure to do that going forward.

I think this is fixed in #3073

**Comment by HalidOdat** [comment]
#3059 Fixes this issue by completely refactoring and redesigning the execution (removed `FinallyStart/End`). I should have assigned the issue to my self to avoid the work overlap :sweat_smile: .

But it's good to have a fix for the panic until #3059 is fully complete :smile:

**Comment by jedel1043** [comment]
> But it's good to have a fix for the panic until #3059 is fully complete 😄

Yeah, and also because we'll probably release tomorrow, so #3059 will probably arrive to our users until 0.18.
